getting-started/setup-multi-node-auth.md

```diff
@@ -66,33 +158,39 @@ Then apply the correct permissions to the file:
 chmod 0600 passfile
 ```
 
-### 3. Reload server configuration
+#### 3. Reload server configuration
 Reload the server configuration on each node for the changes to take effect:
 
 ```bash
 pg_ctl reload
 ```
 
-##### 4. Add users to all nodes
-While connected to the access node:
+#### 4. Attach the data nodes to the access node
 
-```sql
-CREATE ROLE alice WITH LOGIN PASSWORD 'foobar';
-```
+Once the nodes are properly configured, you can continue following the [multi-node setup][attach_multi_node_nodes].
 
-Note: The `postgres` user should already exist, so shouldn't need to be added to each node.
+#### 5. Setting up additional roles [](multi-node-auth-password-roles)
 
-Then, to create the same user on all data nodes, use the `distributed_exec` function 
-([API docs][distributed_exec]) from the access node:
+First, new roles must be added to `passfile` on the access node (create in step 2 above).
+```bash
+*:*:*:testrole:foobar
+```
 
+First, create the role on the access node if needed, and grant it usage to the foreign server
+objects for the data nodes:
 ```sql
-SELECT distributed_exec($$CREATE USER alice WITH PASSWORD 'foobar'$$);
+CREATE ROLE testrole PASSWORD 'foobar';
+GRANT USAGE ON FOREIGN SERVER <data node name>, <data node name>, ... TO testrole;
 ```
 
-and the system setup is complete.
+Finally, use `distributed_exec` to add the role to all of the data nodes. In addition to LOGIN,
```

Comment:
```suggestion
Finally, use [`distributed_exec`][distributed_exec] to add the role to all of the data nodes. In addition to LOGIN,
```

Link these type of things to API docs?

Reply:
Fixed here and in the certificate section